aws/data_source_aws_ecr_authorization_token.go

```diff
@@ -0,0 +1,81 @@
+package aws
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsEcrAuthorizationToken() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEcrAuthorizationTokenRead,
+
+		Schema: map[string]*schema.Schema{
+			"registry_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"authorization_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"proxy_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expires_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"user_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"password": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEcrAuthorizationTokenRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ecrconn
+	params := &ecr.GetAuthorizationTokenInput{}
+	registryId := d.Get("registry_id")
+	if registryId != nil && len(registryId.(string)) > 0 {
+		params.RegistryIds = []*string{aws.String(registryId.(string))}
+	}
+	log.Printf("[DEBUG] Getting ECR authorization token")
+	out, err := conn.GetAuthorizationToken(params)
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("error getting ECR authorization token: %s", err)
+	}
+	log.Printf("[DEBUG] Received ECR AuthorizationData %v", out.AuthorizationData)
+	authorizationData := out.AuthorizationData[0]
+	authorizationToken := *authorizationData.AuthorizationToken
+	expiresAt := strconv.FormatInt(authorizationData.ExpiresAt.Unix(), 10)
+	proxyEndpoint := *authorizationData.ProxyEndpoint
+	authBytes, err := base64.URLEncoding.DecodeString(authorizationToken)
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("error decoding ECR authorization token: %s", err)
+	}
+	basicAuthorization := strings.Split(string(authBytes), ":")
+	userName := basicAuthorization[0]
+	password := basicAuthorization[1]
+	d.SetId(authorizationToken)
```

Comment:
hmm i'm on the fence here as to whether to use the token as the ID, perhaps we should use something like 	`resource.UniqueId()` or a concatenation of non-sensitive fields. what would you recommend @bflad?

Reply:
hi @edgarpoce! asking about this internally, it's suggested the ID here be generated by the TF SDK using the above method call (`resource.UniqueId()`or a timestamp e.g. `d.SetId(time.Now().UTC().String())` as the AWS SDK does not return any id information for us to re-use.

Reply:
fixed in 182dfdb55fe1e3648e7313c15a2509a765eec6c0